Trace history

tcourdy opened the pull request:
Traces now persist across program executions.

Closes #123

rpjohnst committed:
Track the debug session counter

tcourdy committed:
Trace history now persists across program executions

elliothatch reviewed (commented):
It might be better to modify the original DisplayTrace to take a debuggerState, and default to currentDebuggerState if none is passed in, rather than copying creating a new DisplayOldTrace function. That way we don't have to remember to change and test both functions from here on out.

tcourdy committed:
Updates based on elliot's feedback on PR

rpjohnst committed:
Clean up trace history display

- display correct trace count and number
- trace history is usable from function list
